### PR TITLE
fix: show diagram when git errors

### PR DESCRIPTION
**Pull request by noahamac**
This change makes the Diagram meet the documentation spec for non-`deploy` permissioned user behavior. 

https://docs.looker.com/data-modeling/extension-framework/lookml-diagram-admins#granting_permissions_to_use_the_lookml_diagram

The diagram should still be usable for users that do not have git branch metadata. 

Initial report:
```
Hi all, slice life is a heavy user of the LookML diagram and are huge fans. They sent me a note today "It seems like the extension is working now for users with develop permission, but It's not working for users with access_data permission only. 
We were looking at these docs (https://docs.looker.com/data-modeling/extension-framework/lookml-diagram-admins). I have not been able to replicate this locally, but should the extension work with these permissions? 

Here is a screen grab of the permissions below.

https://screenshot.googleplex.com/C5BvQwvLt4hNGLC.png

Please let me know if you have any questions.
```

### CODE REVIEW

**Review by bryans99 — APPROVED** [review]
Code change looks good but remove the eslintcache!

File: src/components/DiagramFrame/DiagramCanvas/DiagramCanvas.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ export const DiagramCanvas: React.FC<DiagramCanvasProps> = ({
   const svgElement = document.querySelector(`svg#display-diagram-svg`)
   const dimensions = currentDimensions?.diagramDict
 
-  if (modelDetail?.fetchError) {
+  if (modelDetail?.fetchError && modelDetail?.fetchError !== 'git') {
```
Comment (bryans99) [inline src/components/DiagramFrame/DiagramCanvas/DiagramCanvas.tsx:116]: Actually, if its a git error what is supposed to happen?

Add some comments?

**Comment by noahamac** [comment]
Ugh, forgot `--staged`
```
noahmac@noahmac-glaptop:~/dev/app-lookml-diagram$ git status
On branch noahamac/git-error-nonfatal
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   .eslintcache
        modified:   src/components/DiagramFrame/DiagramCanvas/DiagramCanvas.tsx

noahmac@noahmac-glaptop:~/dev/app-lookml-diagram$ git restore .eslintcache
```

**Review by bryans99 — APPROVED** [review]
LGTM

**Pull request merged**